base_geoengine/static/src/js/views/geoengine_view.js

```diff
@@ -0,0 +1,508 @@
+ * Odoo geoengine view
+ *---------------------------------------------------------*/
+
+var core = require('web.core');
+var time = require('web.time');
+var View = require('web.View');
+
+var geoengine_common = require('base_geoengine.geoengine_common');
+
+var _lt = core._lt;
+var QWeb = core.qweb;
+
+//var map, layer, vectorLayers = [];
+//TODO: remove this DEBUG
+var map = null;
+var layer = null;
+/* CONSTANTS */
+var DEFAULT_BEGIN_COLOR = "#FFFFFF";
+var DEFAULT_END_COLOR = "#000000";
+var DEFAULT_MIN_SIZE = 5; // for prop symbols only
+var DEFAULT_MAX_SIZE = 15; // for prop symbols only
+var DEFAULT_NUM_CLASSES = 5; // for choroplets only
+var STYLE_DEFAULT = OpenLayers.Util.applyDefaults({
+        fillColor: DEFAULT_BEGIN_COLOR,
+        fillOpacity: 0.8,
+        strokeColor: "#333333",
+        strokeOpacity: 0.8,
+        cursor: "pointer",
+        strokeWidth: 1
+    }, OpenLayers.Feature.Vector.style['default']);
+var STYLE_SELECT = {
+    fillColor: "#ffcc66",
+    strokeColor: "#ff9933"
+};
+
+/**
+ * Method: formatHTML
+ * formats attributes into a string
+ *
+ * Parameters:
+ * a - {Object}
+ */
+var formatHTML = function(a) {
+    var str = [];
+    var oid = '';
+    for (var key in a) {
+        if (a.hasOwnProperty(key)) {
+            var val = a[key];
+            if (val == false) {
+                continue;
+            }
+            var label = '';
+            if (val instanceof Array) {
+                str.push('<span style="font-weight: bold">' + key.charAt(0).toUpperCase() + key.slice(1) + '</span>: ' +val[1]);
+            } else {
+                label = '<span style="font-weight: bold">' + key.charAt(0).toUpperCase() + key.slice(1) + '</span>: ' +val;
+                 if (key == 'id') {
+                    oid = label;
+                } else {
+                    str.push(label);
+                }
+            }
+        }
+    }
+    str.unshift(oid);
+    return str.join('<br />');
+};
+
+var GeoengineView = View.extend(geoengine_common.GeoengineMixin, {
+    template: "GeoengineView",
+    display_name: _lt('Geoengine'),
+    icon: 'fa-map-o',
+
+    init: function(parent, dataset, view_id, options) {
+        this._super(parent);
+        this.set_default_options(options);
+        this.view_manager = parent;
+        this.dataset = dataset;
+        this.model = this.dataset.model;
+        this.view_id = view_id;
+        this.view_type = 'geoengine'
+        this.geometry_columns = {};
+        this.vectorLayers = [];
+
+        // use {selectedFeatures:[]} as a hack to simulate a vector layer
+        // with no feature selected
+        this.selectFeatureControls = [
+            new OpenLayers.Control.SelectFeature(
+            {selectedFeatures:[]}, {hover: true, highlightOnly: true}),
+            new OpenLayers.Control.SelectFeature(
+                {selectedFeatures:[]}, {})
+        ];
+    },
+    limit: function() {
+        return false;
```

Comment:
This return was already there in 8.0. Was it for debug ? Not sure we even need limit there. Thus I commented code for clarity and to remove Warning message.

@lmignon any reason for this ?
:smiling_imp: https://github.com/OCA/geospatial/commit/915925e206f81b1caf409323df6cd6b71be54278#diff-f6f3909c744c4f386dfa5d5cc1d99cccR171